src/Controls/src/Core/TypeConversionHelper.cs

```diff
@@ -0,0 +1,172 @@
+
+			if (value is IWrappedValue { Value: var wrappedValue, ValueType: var wrappedType })
+			{
+				if (wrappedValue is null)
+				{
+					convertedValue = null;
+					return !targetType.IsValueType || targetType.IsGenericType && targetType.GetGenericTypeDefinition() == typeof(Nullable<>);
+				}
+
+				// It might be enough just to unwrap the value
+				if (targetType.IsAssignableFrom(wrappedType))
+				{
+					convertedValue = wrappedValue;
+					return true;
+				}
+
+				value = wrappedValue;
+				valueType = wrappedType;
+			}
+
+			if (TryGetTypeConverter(valueType, out var converter) && converter is not null && converter.CanConvertTo(targetType))
+			{
+				convertedValue = converter.ConvertTo(value, targetType) ?? throw new InvalidOperationException($"The {converter.GetType()} returned null when converting {valueType} to {targetType}");
+				return true;
+			}
+
+			if (TryGetTypeConverter(targetType, out converter) && converter is not null && converter.CanConvertFrom(valueType))
+			{
+				convertedValue = converter.ConvertFrom(value) ?? throw new InvalidOperationException($"The {converter.GetType()} returned null when converting from {valueType}");
+				return true;
+			}
+
+			if (RuntimeFeature.IsImplicitCastOperatorsUsageViaReflectionSupported)
+			{
+				if (TryConvertUsingImplicitCastOperator(value, targetType, out convertedValue))
+				{
+					return true;
+				}
+			}
+			else
+			{
+				WarnIfImplicitOperatorIsAvailable(value, targetType);
+			}
+
+			convertedValue = null;
+			return false;
+		}
+
+		private static bool TryGetTypeConverter(Type type, [NotNullWhen(true)] out TypeConverter? converter)
+			=> TypeConversionAppBuilderExtensions.TryGetTypeConverter(type, out converter)
+				|| type.TryGetTypeConverter(out converter);
+
+		[RequiresUnreferencedCode("The method uses reflection to find implicit conversion operators. " +
+			"It is not possible to guarantee that trimming does not remove some of the implicit operators. " +
+			"Consider preserving op_Implicit methods through DynamicDependency or DynamicallyAccessedMembers attributes.",
+			Url = "https://learn.microsoft.com/dotnet/core/deploying/trimming/prepare-libraries-for-trimming")]
+		private static bool TryConvertUsingImplicitCastOperator(object value, Type targetType, [NotNullWhen(true)] out object? result)
+		{
+			Type valueType = value.GetType();
+			MethodInfo? opImplicit = GetImplicitConversionOperator(valueType, fromType: valueType, toType: targetType)
+										?? GetImplicitConversionOperator(targetType, fromType: valueType, toType: targetType);
+
+			object? convertedValue = opImplicit?.Invoke(null, new[] { value });
+			if (convertedValue is not null)
+			{
+				result = convertedValue;
+				return true;
+			}
+
+			result = null;
+			return false;
+
+			[UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:RequiresUnreferencedCode",
+				Justification = "We add our own RequiresUnreferencedCode attribute to the parent method.")]
```

Comment:
This should be RUC as well - since it's called only from a RUC method, there will be no warning produced, but it will guard against potential future changes which might call this method from somewhere else.